horde_worker_regen/process_management/process_manager.py

```diff
@@ -2752,65 +2767,70 @@ async def _get_source_images(self, job_pop_response: ImageGenerateJobPopResponse
+                if extra_source_image.image.startswith("http"):
+                    any_extra_source_images_are_urls = True
+                    break
 
-        for field in image_fields:
-            try:
-                field_value = new_response_dict[field]
-                if field_value is not None:
-                    image = PIL.Image.open(BytesIO(base64.b64decode(field_value)))
-                    image.verify()
+        attempts = 0
+        while attempts < MAX_SOURCE_IMAGE_RETRIES:
+            if (
+                source_image_is_url
+                and job_pop_response.source_image is not None
+                and job_pop_response.get_downloaded_source_image() is None
+            ):
+                download_tasks.append(job_pop_response.async_download_source_image(self._aiohttp_client_session))
+            if (
+                source_mask_is_url
+                and job_pop_response.source_mask is not None
+                and job_pop_response.get_downloaded_source_mask() is None
+            ):
+                download_tasks.append(job_pop_response.async_download_source_mask(self._aiohttp_client_session))
 
-            except Exception as e:
-                logger.error(f"Failed to verify {field}: {e}")
-                if job_pop_response.id_ is None:
-                    raise ValueError("job_pop_response.id_ is None") from e
+            download_extra_source_images = job_pop_response.get_downloaded_extra_source_images()
+            if (
+                any_extra_source_images_are_urls
+                and job_pop_response.extra_source_images is not None
+                or (
+                    download_extra_source_images is not None
+                    and job_pop_response.extra_source_images is not None
+                    and len(download_extra_source_images) != len(job_pop_response.extra_source_images)
+                )
+            ):
 
-                new_meta_entry = GenMetadataEntry(
-                    type=METADATA_TYPE[field],
-                    value=METADATA_VALUE.parse_failed,
+                download_tasks.append(
+                    asyncio.create_task(
+                        job_pop_response.async_download_extra_source_images(
+                            self._aiohttp_client_session,
+                            max_retries=MAX_SOURCE_IMAGE_RETRIES,
+                        ),
+                    ),
                 )
-                self.job_faults[job_pop_response.id_].append(new_meta_entry)
 
-                new_response_dict[field] = None
-                new_response_dict["source_processing"] = "txt2img"
+            gather_results = await asyncio.gather(*download_tasks, return_exceptions=True)
 
-        return ImageGenerateJobPopResponse(**new_response_dict)
+            for result in gather_results:
+                if isinstance(result, Exception):
+                    logger.error(f"Failed to download source image: {result}")
```

Comment:
We need to add the relevant `GenMetadataEntry` for download and parse errors as well